mm2src/coins/utxo/utxo_builder/utxo_arc_builder.rs

```diff
@@ -375,6 +386,203 @@ pub(crate) async fn block_header_utxo_loop<T: UtxoCommonOps>(
+            };
+
+            // If electrum returns response too large error, we will reduce the requested headers by CHUNK_SIZE_REDUCER_VALUE in every loop until we arrive at a reasonable value.
+            if err_inner.is_response_too_large() && *chunk_size > CHUNK_SIZE_REDUCER_VALUE {
+                *chunk_size -= CHUNK_SIZE_REDUCER_VALUE;
+                return Err(UtxoLoopCtrlStatement::Continue(err.to_string()));
+            }
+
+            if *fetch_blocker_headers_attempts > 0 {
+                *fetch_blocker_headers_attempts -= 1;
+                error!(
+                    "Error {err:?} on retrieving latest {ticker} headers from rpc! \
+                    {fetch_blocker_headers_attempts} attempts left"
+                );
+                // Todo: remove this electrum server and use another in this case since the headers from this server can't be retrieved
+                return Err(UtxoLoopCtrlStatement::Continue(err.to_string()));
+            };
+
+            error!(
+                "Error {err:?} on retrieving latest {ticker} headers from rpc after {FETCH_BLOCK_HEADERS_ATTEMPTS} \
+                    attempts"
+            );
+            // Todo: remove this electrum server and use another in this case since the headers from this server can't be retrieved
+            return Err(UtxoLoopCtrlStatement::Break(err.to_string()));
+        },
+    };
+
+    Ok((block_registry, block_headers))
+}
+
+/// Retrieves block headers from the specified client within the given height range and revalidate against [`SPVError::ParentHashMismatch`] .
+///
+/// If the height is equal to the starting block height, the loop will break.
+/// Otherwise, it will attempt to fetch the block headers up to the specified `to_height`
+/// It will continue to attempt fetching headers until there's no [`SPVError::ParentHashMismatch`] or runs out of
+/// attempts.
+///
+/// If there is an error during header retrieval, it will check the error type and
+/// take appropriate action based on whether it is a temporary or permanent error.
+///
+/// # Arguments
+///
+/// * `coin` - The coin to retrieve block headers for.
+/// * `client` - The client to use for retrieving block headers.
+/// * `height` - The starting block height for header retrieval.
+/// * `to_height` - The ending block height for header retrieval.
+/// * `chunk_size` - The size of the chunks to retrieve headers in.
+/// * `sync_status_loop_handle` - The handle to notify in case of errors.
+/// * `args` - The arguments for the utxo loop.
+///
+/// # Returns
+///
+/// A Result `Ok(())` or `UtxoLoopCtrlStatement` if there was a break or continue instruction.
+async fn retrieve_and_revalidate_mismatching_header(
+    sync_status_loop_handle: &mut UtxoSyncStatusLoopHandle,
+    storage: &dyn BlockHeaderStorageOps,
+    (client, spv_conf): (&ElectrumClient, &SPVConf),
+    (coin, mut from_height): (&str, u64),
+    chunk_size: &mut u64,
+) -> Result<(), UtxoLoopCtrlStatement> {
```

Comment:
you shouldn't include unrelated tuples in a function signature as you have done here. This is a bad practice.

Reply:
Sorry didn't take that into consideration..was try to avoid the compiler screaming for too many args